src/Datasource/RepositoryInterface.php

```diff
@@ -115,26 +115,46 @@ public function updateAll($fields, $conditions);
     public function deleteAll($conditions);
 
     /**
-     * Delete each record individually by matching the conditions. For each
-     * matched record Table::delete() is executed providing support for behaviors
-     * and events.
+     * Deletes records individually by reading batches that match the conditions. For each
+     * matched record Table::delete() is executed providing support for behaviors and events.
      *
-     * This method will continue execution until all records are deleted. You should
-     * be aware of possible long execution times, and side effects of processing
-     * events on a large collection of records.
+     * This method will continue execution until all records are deleted, or one of the events is stopped.
+     * If a record fails to be deleted the operation will stop (optional).
+     *
+     * This method will *not* trigger afterDeleteCommit events. The event afterDeleteEachCommit will be
+     * broadcast after each batch is committed.
+     *
+     * This method will *not* accept query options that alter the size of a batch, such as limit, offset
+     * and page.
      *
      * ### Options
      *
-     * Options are passed to Table::delete() with the exception of these.
+     * - 'finder' string Defaults to 'all'. The customer finder to use when reading batches.
+     * - `atomic` boolean Defaults to true. When true the deletion happens within a transaction.
+     * - 'stopOnFailure' boolean Defaults to true. Stop deleting records when a record fails to be deleted.
+     * - 'batch' integer Defaults to 1. Number of records to read in batches, zero reads all records.
+     * - `checkRules` boolean Defaults to true. Check deletion rules before deleting the record.
```

Comment:
Personally I would still prefer to let the IDE help (autocomplete and existence check) with class constants for the option keys ;)
`RepositoryInterface::FINDER => ...`